src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs

```diff
@@ -713,7 +713,15 @@ internal static void ManagePortConnectionEvents(Step stepInfo, StepUIAutomation
 
         private static void OutputPortViewModel_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
         {
-            var buttonName = CurrentExecutingStep.UIAutomation[2].Parameters[0] as string;
+            if (CurrentExecutingStep.UIAutomation == null) return;
+
+            var currentUiAutomation = CurrentExecutingStep.UIAutomation.FirstOrDefault(x => x.Sequence == 3);
```

Comment:
This still seems a bit hardcoded specifically for this step, is there a way to make this function not specific to this step?

Reply:
I updated the code by searching for its method name. I guess in this way it will be less hardcoded. Please, let me know your opinion.